CharlieBackend.Api.UnitTes/ScheduleServiceTests.cs

```diff
@@ -0,0 +1,412 @@
+            //Arrange
+            var createScheduleDto = new CreateScheduleDto
+            {
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.Daily,
+                    Interval = 1
+                },
+
+                Range = new OccurenceRange { },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = existentGroupId
+                }
+            };
+
+            _studentGroupRepositoryMock.Setup(x => x.IsEntityExistAsync(1)).ReturnsAsync(false);
+
+            Initialize(createScheduleDto);
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.CreateScheduleAsync(createScheduleDto);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_NotValidMentorId_ShouldReturnValidationError()
+        {
+            //Arrange
+            var createScheduleDto = new CreateScheduleDto
+            {
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.Daily,
+                    Interval = 1
+                },
+
+                Range = new OccurenceRange { },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = existentGroupId,
+                    MentorID = existentMentorId
+                }
+            };
+
+            _studentGroupRepositoryMock.Setup(x => x.IsEntityExistAsync(existentGroupId)).ReturnsAsync(true);
+
+            _mentorRepositoryMock.Setup(x => x.IsEntityExistAsync(existentMentorId)).ReturnsAsync(false);
+
+            Initialize(createScheduleDto);
+
+            var scheduleService = new ScheduleService(_unitOfWorkMock.Object, _mapper, _scheduledEventFactory);
+
+            //Act
+            var result = await scheduleService.CreateScheduleAsync(createScheduleDto);
+
+            //Assert
+            result.Should().NotBeNull();
+            result.Error.Code.Should().BeEquivalentTo(ErrorCode.ValidationError);
+        }
+
+        [Fact]
+        public async Task CreateScheduleAsync_NotValidThemeId_ShouldReturnValidationError()
+        {
+            //Arrange
+            var createScheduleDto = new CreateScheduleDto
+            {
+                Pattern = new PatternForCreateScheduleDTO
+                {
+                    Type = PatternType.Daily,
+                    Interval = 1
+                },
+
+                Range = new OccurenceRange { },
+
+                Context = new ContextForCreateScheduleDTO
+                {
+                    GroupID = existentGroupId,
+                    MentorID = existentMentorId,
+                    ThemeID = -1
```

Comment:
Same as here: https://github.com/ita-social-projects/WhatBackend/pull/406#discussion_r581908218
Perhaps we could introduce a `nonExistentThemeId` to avoid using magic numbers?

Reply:
Done